src/sonic-config-engine/minigraph.py

```diff
@@ -615,34 +726,59 @@ def parse_xml(filename, platform=None, port_config_file=None):
+            if asic_name is None:
+                (intfs, lo_intfs, mvrf, mgmt_intf, vlans, vlan_members, pcs, pc_members, acls, vni) = parse_dpg(child, hostname)
+            else:
+                (intfs, lo_intfs, mvrf, mgmt_intf, vlans, vlan_members, pcs, pc_members, acls, vni) = parse_dpg(child, asic_name)
         elif child.tag == str(QName(ns, "CpgDec")):
-            (bgp_sessions, bgp_asn, bgp_peers_with_range, bgp_monitors) = parse_cpg(child, hostname)
+            if asic_name is None:
+                (bgp_sessions, bgp_asn, bgp_peers_with_range, bgp_monitors) = parse_cpg(child, hostname)
+            else:
+                (bgp_sessions, bgp_asn, bgp_peers_with_range, bgp_monitors) = parse_cpg(child, asic_name)
         elif child.tag == str(QName(ns, "PngDec")):
-            (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speed_png, console_ports) = parse_png(child, hostname)
+            if asic_name is None:
+                (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speed_png, console_ports, device_neighbors) = parse_png(child, hostname)
+            else:
+                (neighbors, devices, port_speed_png, device_neighbors) = parse_asic_png(child, asic_name, hostname)
         elif child.tag == str(QName(ns, "UngDec")):
-            (u_neighbors, u_devices, _, _, _, _, _, _) = parse_png(child, hostname)
+            (u_neighbors, u_devices, _, _, _, _, _, _, _) = parse_png(child, device_hostname)
         elif child.tag == str(QName(ns, "MetadataDeclaration")):
-            (syslog_servers, dhcp_servers, ntp_servers, tacacs_servers, mgmt_routes, erspan_dst, deployment_id, region) = parse_meta(child, hostname)
+            if asic_name is None:
+                (syslog_servers, dhcp_servers, ntp_servers, tacacs_servers, mgmt_routes, erspan_dst, deployment_id, region) = parse_meta(child, hostname)
+            else:
+                (deployment_id, sub_role) = parse_asic_meta(child, asic_name)
         elif child.tag == str(QName(ns, "DeviceInfos")):
             (port_speeds_default, port_descriptions) = parse_deviceinfo(child, hwsku)
 
-    current_device = [devices[key] for key in devices if key.lower() == hostname.lower()][0]
+    if asic_name is None:
+        current_device = [devices[key] for key in devices if key.lower() == hostname.lower()][0]
+        name = hostname
+    else:
+        current_device = [devices[key] for key in devices if key.lower() == asic_name.lower()][0]
+        name = asic_name
+
     results = {}
     results['DEVICE_METADATA'] = {'localhost': {
         'bgp_asn': bgp_asn,
         'deployment_id': deployment_id,
         'region': region,
         'docker_routing_config_mode': docker_routing_config_mode,
-        'hostname': hostname,
+        'hostname': name,
         'hwsku': hwsku,
-        'type': current_device['type']
+        'type': current_device['type']   
```

Comment:
remove trailing space. please enable your editor to remove trailing space automatically. it is annoy to see such trailing space here.

Reply:
Fixed